### ISSUE TITLE
remove support for node 6

### ISSUE THREAD

**Issue by hildjj** [hildjj/node-cbor#95]
We have now pinned an old version of [ava](https://github.com/avajs/ava) so I can still test on Node 6.  There will be other reasons to move over time.

This is a tracking bug.  Please reference this with a bug in your project if you want me to wait for you to stop supporting Node 6.

**Comment by hildjj** [comment]
Blocks full fix for #10.

**Comment by hildjj** [comment]
I'm thinking about adding a warning in the postinstall hook that you should upgrade node.  I'll do a major version for that, and everyone that wants to support node < v10 will need to pin to the minor version before that.  Ava now has an audit alarm on the version I'm forced to use if I'm going to test on early node versions, and I don't want to claim to support those versions if I can't test on them.

**Comment by hildjj** [comment]
This is becoming more pressing for me as security issues continue to pile up for old versions of ava.  If someone wanted to do the research to pick a test framework that works on node6, send me a patch that moves to that framework, and ensures coverage doesn't suffer too badly, I'd take it.

**Comment by hildjj** [comment]
OK, my current approach is that `package.json`'s `engines` entry doesn't do anything for older versions of npm.  `engineStrict` is [deprecated](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#enginestrict).  I've added a `.npmrc` file with [`engine-strict`](https://docs.npmjs.com/cli/v6/using-npm/config#engine-strict)`=true`, but that probably only works for newer versions of npm.  In the future, it may be enough.  In the meantime, I've added an aggressive preinstall hook that will check the current node version.  You'll get a warning like:

```
+----------------------------------------------------+
| The cbor package REQUIRES Node.js 10.18 or higher. |
| Please upgrade node, or pin cbor to version 5.2.x: |
| `npm install --save cbor@5.2`                      |
+----------------------------------------------------+
```

I'm open to taking patches to backport features or bug fixes into the 5.2 branch (which I've tagged `node6`, and which tag I'll move forward with changes), but testing is going to be iffy because of the old version of ava involved. So... sparingly, I guess.

Please just upgrade to a [supported](https://github.com/nodejs/Release#release-schedule) version of Node and encourage your distros and users to do so also.
